src/syntax/ppx_sedlex.ml

```diff
@@ -311,16 +458,37 @@ let regexp_of_pattern env =
       | _ ->
           err p.ppat_loc @@ "the " ^ name
           ^ " operator requires two arguments, like " ^ name ^ "(a,b)"
-  and aux p =
+  and aux ?(allow_alias = false) p =
+    let loc = p.ppat_loc in
     (* interpret one pattern node *)
     match p.ppat_desc with
-      | Ppat_or (p1, p2) -> Sedlex.alt (aux p1) (aux p2)
+      | Ppat_or (p1, p2) ->
+          let r1, s1 = aux ~allow_alias p1 in
+          let r2, s2 = aux ~allow_alias p2 in
+          if not (StrLocSet.equal s1 s2) then begin
```

Comment:
Seems to be a mistake to use StrLocSet to check if the two side of or-pattern define the same aliases

Reply:
Yes. I'll fix it later.